### ISSUE TITLE
Custom parameters in New Online Repo form

### ISSUE THREAD

**Issue by ilpelatodijavascript** [abapGit/abapGit#5345]
Hello everyone.
My requirement is to add a few parameters to the "New Online Repo" form.
I've seen the documentation but it is not very clear.
I already tried to create a subclass of ZCL_ABAPGIT_GUI_PAGE and redefine RENDER_CONTENT with the example provided just to see where the text would be placed, but it is not showed anywhere and I can't even find it in the html code of the page.
Is this possible?
Thanks

**Comment by larshp** [comment]
custom parameters? can you elaborate a bit more on the scenario?

**Comment by ilpelatodijavascript** [comment]
Sure. I just need to add a few checkboxes that will create an entry on a Z table.
This stuff is not related to abapgit logic at all.

**Comment by larshp** [comment]
consider using https://docs.abapgit.org/ref-exits.html#wall_message_repo to add a message like "It is required to add an entry to Z table", assuming it does not exist

its still a bit unclear to me what the requirement is, if you need additional information in the git repository, add that in the git repository on the git host. If it is related to packages on the ABAP system, it will still be a problem outside of abapGit?

**Comment by larshp** [comment]
information regarding the creator and last "puller" can be shown in the overview by clicking details, 

![image](https://user-images.githubusercontent.com/5888506/153653748-b39b91ad-986a-47eb-ae09-d9bff0ee35f2.png)

**Comment by ilpelatodijavascript** [comment]
> consider using https://docs.abapgit.org/ref-exits.html#wall_message_repo to add a message like "It is required to add an entry to Z table", assuming it does not exist

I guess it will work too.
Thanks

**Comment by ilpelatodijavascript** [comment]
Hi, reopening because I got an idea.

I was not able to get in debug on the ON_EVENT exit so I’m not quite sure if this already happens, but how about trigger that when user goes into a page?
Then we can have in the parameters the html page where it can be modified.

**Comment by larshp** [comment]
fair enough, but can you share some details on the scenario? as an open source project we dont want to support/take into account any scenario. If there is a specific scenario, we would like to understand it before/if implementing support for it.

**Comment by ilpelatodijavascript** [comment]
Maybe my own requirement is not really the best example for this scenario, but I can see it being useful for something like default values of pages (prefill the repository name, url or package field with a prefix) etc.

Is this a scenario you think people need?

**Comment by larshp** [comment]
okay

default values of pages: do you want to restrict the url's to only some pattern? Being able to pull code from anywhere is both good and bad, but your internal quality assurance must, in any case, ensure that the changes are okay.

or alternatively, are you talking about a feature where, from a url, the package name is defaulted?

**Comment by ilpelatodijavascript** [comment]
That is not my requirement however a solution like the ON_EVENT triggered when the page is opened with the html as parameter should allow you to do both of the scenarios you said, and I can see both of them being really useful (especially a regex pattern on the repository url field).

So from the html object parameter of the exit, get the repository url field and add validation.
However, at the time, this is not possible right?

Just an idea of scenario where a modified ON_EVENT exit modified could be useful.

**Comment by larshp** [comment]
if its a scenario of not allowing connections, then my suggestion is to not allow any connections from the SAP system to the HTTP destination

perhaps its possible to implement custom logic, but that custom logic will not prevent establishing the connection

if you defiantly want to do this, try throwing an exception in https://docs.abapgit.org/ref-exits.html#wall_message_repo, or do ASSERT to trigger a short dump

**Comment by ilpelatodijavascript** [comment]
Ok so that could be done by blocking the http connection, but how would you be able to automatically fill the package field based on the repo url?

In this case, isn’t the exit with a modified page the best solution? I would try to add a js script to it.

**Comment by larshp** [comment]
well, it depends on what you are trying to solve/what the scenario is...

a JS script will not block the connection as such, if you want to block the connection, then I suggest blocking the connection

**Comment by larshp** [comment]
blocking the connection: if you want to block the connection, then you want to block it for every connection? and not for one particular application? ie. it should be blocked on server level, not application level.

**Comment by larshp** [comment]
if you want to prevent code from getting into your system: its the same problem as if you have any developer add code to your system

but again, I'm just guessing as to what your scenario is/what you are trying to solve, can you [elaborate a bit more on the scenario](https://github.com/abapGit/abapGit/issues/5345#issuecomment-1036512436). As an open source project we don't want to support/take into account any scenario. If there is a specific scenario, we would like to understand it before/if implementing support for it.

**Comment by mbtools** [comment]
Hi @ilpelatodijavascript 

if you are using the full developer version of abapGit, you can replace the "New Online" page with your own custom version. 

1. Copy `zcl_abapgit_gui_page_addonline` to your own class `zcl_your_gui_page_addonline`
2. Implement `on_event` exit and redirect action `zif_abapgit_definitions=>c_action-repo_newonlne` to your new class

```abap
      WHEN zif_abapgit_definitions=>c_action-repo_newonline.                  
        rs_handled-page  = zcl_your_gui_page_addonline=>create( ).
        rs_handled-state = zcl_abapgit_gui=>c_event_state-new_page.
    ENDCASE.
```

3. Enhance your new class as you wish

https://docs.abapgit.org/ref-exits.html#on_event

I hope this helps. If you have requirements that are relevant for abapGit, please open a new issue with a detailed description of your use case.

Thanks,
Marc
